encoding/box_query_encoding.go

```diff
@@ -0,0 +1,43 @@
+package encoding
+
+import (
+	"encoding/base64"
+	"github.com/algorand/go-algorand-sdk/client/v2/common/models"
+	"github.com/algorand/go-algorand-sdk/types"
+)
+
+const encodingBase64Prefix = "b64:"
+
+// EncodeBytesForBoxQuery provides a convenience method to string encode box names for use with Box search APIs (e.g. GetApplicationBoxByName).
+func EncodeBytesForBoxQuery(xs []byte) string {
+	return encodingBase64Prefix + base64.StdEncoding.EncodeToString(xs)
+}
+
+// EncodeBoxForBoxQuery provides a convenience method to string encode box names for use with Box search APIs (e.g. GetApplicationBoxByName).
+func EncodeBoxForBoxQuery(b models.Box) (string, error) {
+	decoded, err := base64.StdEncoding.DecodeString(string(b.Name))
+	if err != nil {
+		return "", err
+	}
+
+	return EncodeBytesForBoxQuery(decoded), nil
+}
+
+func EncodeBoxDescriptorForBoxQuery(bd models.BoxDescriptor) (string, error) {
+	decoded, err := base64.StdEncoding.DecodeString(string(bd.Name))
```

Comment:
I believe that `bd.Name` will already be base64 decoded by the JSON decoder.

This is not obvious, since the `Name` field still retains the REST API description that it's a base64 encoded string. Yet that's no longer accurate here.